### PR TITLE
Added a check for AMI encryption

### PR DESCRIPTION
**Pull request by adamjohnson01**
Fix for #858

I have added a check to see if the ami being used is encrypted and updated the cloudformation template to include the value of that check. This is required to enable the use of encrypted amis

- [x] Code compiles correctly (i.e `make build`)
- [x] Added tests that cover your change (if possible)
- [x] All unit tests passing (i.e. `make test`)
- [x] Manually tested

### CODE REVIEW

File: pkg/cfn/builder/api_test.go [hunk 1]
```diff
@@ -297,6 +297,8 @@ var _ = Describe("CloudFormation template builder API", func() {
 		*ng.VolumeType = api.NodeVolumeTypeIO1
 		ng.VolumeName = new(string)
 		*ng.VolumeName = "/dev/xvda"
+		ng.VolumeEncrypted = new(bool)
```
[the remark this answers is not in the document]
Comment (adamjohnson01) [reply pkg/cfn/builder/api_test.go]: Sure, will do. Thanks!

File: pkg/apis/eksctl.io/v1alpha5/types.go [hunk 2]
```diff
@@ -406,6 +406,9 @@ type NodeGroup struct {
 	// +optional
 	VolumeName *string `json:"volumeName,omitempty"`
 	// +optional
+	VolumeEncrypted *bool `json:"volumeEncrypted,omitempty"`
+	//
```
[the remark this answers is not in the document]
Comment (adamjohnson01) [reply pkg/apis/eksctl.io/v1alpha5/types.go]: Will remove it, thanks!

File: pkg/cfn/builder/api_test.go [hunk 3]
```diff
@@ -297,6 +297,8 @@ var _ = Describe("CloudFormation template builder API", func() {
 		*ng.VolumeType = api.NodeVolumeTypeIO1
 		ng.VolumeName = new(string)
 		*ng.VolumeName = "/dev/xvda"
+		ng.VolumeEncrypted = api.Disabled()
+		*ng.VolumeEncrypted = false
```
Comment (errordeveloper) [inline pkg/cfn/builder/api_test.go]: This is redundant

Comment (adamjohnson01) [reply pkg/cfn/builder/api_test.go]: I have removed. Thanks!

**Comment by adamjohnson01** [comment]
> Thank you for the PR! Looks great! Can you add a little test that shows that when we enable it the right field appear in the cloud formation resource? Perhaps just switching the existing test to true and checking that the launchtemplate is correct?

@martina-if , I have added a new test. I have also split out the Ebs tests that were already there into their own test. I hope that is OK.

**Comment by errordeveloper** [comment]
@adamjohnson01 could you please rebase, run `make generate-kubernetes-types` and commit updated files? After that we can merge.

**Comment by martina-if** [comment]
@adamjohnson01 even better. This looks great, thanks!

**Comment by errordeveloper** [comment]
I'll merge and run `make generate-kubernetes-types` later.

**Pull request merged**